mesonbuild/interpreter/interpreter.py

```diff
@@ -296,9 +296,8 @@ def __init__(
         self.configure_file_outputs: T.Dict[str, int] = {}
         # Passed from the outside, only used in subprojects.
         if default_project_options:
-            self.default_project_options = default_project_options if isinstance(default_project_options, str) else default_project_options.copy()
-            if isinstance(default_project_options, dict):
-                pass
+            assert isinstance(default_project_options, dict)
+            self.default_project_options = default_project_options
```

Comment:
This is erasing the copy because it's getting converted back to a list wherever used, correct?

Reply:
Yes and it should be the caller's knowledge whether it can give ownership or a copy is needed